### PR TITLE
chore(core): Update to Node 12.14.1

### PR DESCRIPTION
**Pull request by anotherchrisberry**
* use the latest LTS versions of node, npm, yarn
* pin fsevents to 1.2.11 (node compatibility issue)
* update webdriver to v5 (node compatibility, lots of breaking changes there)
* update a bunch of other dependencies, which were just things I changed trying to get things working ~before @christopherthielen recommended the `resolutions` trick~.

The webdriver part was kind of a pain, since the APIs changed a fair amount and I haven't worked with webdriver in years, and I'd never looked at the code in this repo around it.

### CODE REVIEW

File: wdio.conf.js [hunk 1]
```diff
@@ -64,13 +66,15 @@ const config = {
...
+  beforeSession: (_config, _capabilities, specs) => {
+    if (!specs.length) {
+      return;
+    }
     if (!flags['replay-fixtures'] && !flags['record-fixtures']) {
       return;
     }
     const fixtureService = new FixtureService();
-    testRun = { fixtureFile: fixtureService.fixturePathForTestPath(test.file) };
+    testRun = { fixtureFile: fixtureService.fixturePathForTestPath(specs[0]) };
```
Comment (anotherchrisberry) [inline wdio.conf.js]: We used to have a hook into test file names via the `beforeTest` method; however, the API has changed, and `test` no longer include file information. We are able to get it in the `beforeSession` call, as there's always just one entry in `specs`, and it happens to be the absolute file name of the test.

It's possible this doesn't work in some scenario, but it works in all the ones we have, which is good enough for me?

File: gradle/installViaTravis.sh [hunk 2]
```diff
@@ -15,7 +15,7 @@ if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
 
   source ~/.nvm/nvm.sh
   nvm install $NODE_JS_VERSION
-  yarn --frozen-lockfile
+  yarn --ignore-optional --frozen-lockfile
```
Comment (anotherchrisberry) [inline gradle/installViaTravis.sh]: I don't like this. Adding a version of `fsevents` to `resolutions` seems to be overriding the `optionalDependencies` entry. 

I'm going to rework this. We don't want to have to explain to non-Mac users that they have to add `--ignore-optional` when running `yarn`.

**Comment by anotherchrisberry** [comment]
Updated to use Node 12.14.1 with new GHA

**Comment by anotherchrisberry** [comment]
@christopherthielen updated comment; short version: `resolutions` wasn't necessary and caused other problems.

The transitive dependencies that include `fsevents` all declare it in the `optionalDependencies` block, but, when I added it to `resolutions`, Travis builds all tried to install it and failed the build (since they run on Linux and `fsevents` is a Mac-only library).

There's a commit (later reverted) in this branch that added an `--ignore-optional` flag to the Travis build, but I felt like that would be bad for any non-Mac users if they suddenly had to add that flag to their `yarn` commands.

So, I did something that felt weird and wrong but ended up being fine: I went into the `yarn.lock` file, deleted the entry for `fsevents`, then re-ran `yarn`. Yarn regenerated the entry in the lock file, but resolved `fsevents` to the explicitly-declared version in deck's `optionalDependencies` block, and everything was fine.

I don't understand *why* this fixed it, but it did.

**Comment by christopherthielen** [comment]
lovely, thanks for the explanation.  
I deleted travis.

**Comment by anotherchrisberry** [comment]
@christopherthielen @maggieneterval any concerns merging this? it will require folks to upgrade their version of node, and potentially yarn. It might be enough to let people know in OSS Slack.

**Comment by maggieneterval** [comment]
> it will require folks to upgrade their version of node, and potentially yarn. It might be enough to let people know in OSS Slack

The node upgrade will only affect folks developing locally or building a custom docker image, right? Seems sufficient to send out a Slack message to #deck and #dev :+1:

**Pull request merged**